EmbeddedPkg/Library/VirtualRealTimeClockLib/VirtualRealTimeClockLib.c

```diff
@@ -103,7 +103,7 @@ LibGetTime (
   }
 
   Counter       = GetPerformanceCounter ();
-  EpochSeconds += DivU64x64Remainder (Counter, Freq, &Remainder);
+  EpochSeconds += (UINTN)DivU64x64Remainder (Counter, Freq, &Remainder);

```

Comment:
This might address the codeql error, but its losing information because its casting to truncate.

have you considered changing EpochSeconds to a UINT64, so that data is not lost?

Reply:
You are right, in 32-bits system, UINT64 and UINTN are different.
To make aligned in 32-bits and 64-bits system, UINT64 as base data type is more suitable.